### DISCUSSION TITLE
Import Improvements

### DISCUSSION THREAD

**Discussion by MikeBishop** [actualbudget/releases#21] [Feature ideas]
While we all want to see Direct Import sooner than later, I had a few smaller requests for file-based import:

- Ignore transactions which occur prior to the Starting Balance entry
- Identify transactions which have been imported previously and don't attempt to import them again
- Consider supporting drag-and-drop of the file onto the Actual window and/or registering Actual as the program to use to open import file types.  (Likely requires prompting for the account to import to.)

**Comment by Erystiva** [comment]
An option to invert payment and deposit would also be useful - this has been an issue I've encountered with various other budgeting apps including YNAB3 and YNAB4 where the credit card export file (.qif) for my bank imports with the credit card inflow and outflow is reversed. MoneyWiz and nYNAB overcame this by giving the user the opportunity to swap the columns.

With Actual I guess the fix would be having a option to switch negative values to positive, positive to negative on the Import Transactions window.

**Comment by stephmeyer** [comment]
Yes, I already noticed this is an issue for me too, with file importing.  I would have to export only the dates I want from the bank, as it's not an option to just ignore some of the transactions, I would instead have to manually delete them once they import.  Definitely worth adding and probably not too much extra work to implement, compared to direct importing.

**Reply by bpaulien** [reply]
I don't have a problem with selecting date range of export on bank's site (although I suppose some banks don't allow for that? Not sure, as all of mine do) but I definitely would like the drag/drop option to just pull the file over after downloading, and let it automatically import.

**Comment by Medicliffy** [comment]
I'd also suggest giving the option to ignore the notes field when importing files.

Some of my banks put payee data there that isn't useful to me (without the option to exclude it), and the favoring of the imported transactions (which in all other cases makes sense) deletes any notes I have made when manually inputting transactions. I'm quite new to Actual, so I've not figured out a good workaround - might just do it manually or use CSV import (though that has its own issues because the same bank puts the credits and debits in separate columns...gotta love some export formats).

**Reply by smathson** [reply]
I think this specific issue was actually a regression in the last release and should be fixed in the next one. Any notes on existing transactions should be respected during import.
